components/mintKamon/GenerateImageForm.tsx

```diff
@@ -1,13 +1,221 @@
+        toastDescription += 'Please enter your name. '
+      }
+      if (!imageFile) {
+        toastDescription += 'Please choose a profile picture. '
+      } else if (!imageFileObject) {
+        toastDescription +=
+          'Your profile picture could not be correctly processed. '
+      }
+      toast({
+        title: 'All fields are required',
+        description: toastDescription,
+        status: 'error'
+      })
+      return
+    }
+    setIsLoading(true)
+    // this is debug purpose
+    // setTokenImageURI(
+    //   'https://pitpa.mypinata.cloud/ipfs/QmNndSLYnChANwhPBnj5AwB5M7XEWz5LwDbQsrXNLu89Gb'
+    // )
+    const blobToBase64 = (blob: any): Promise<string> => {
+      return new Promise<string>((resolve, _) => {
+        const reader = new FileReader()
+        reader.onloadend = () => {
+          if (typeof reader.result == 'string') {
+            resolve(reader.result)
+          } else {
+            resolve('')
+          }
+        }
+        reader.readAsDataURL(blob)
+      })
+    }
+
+    const payLoad = {
+      name,
+      address: data?.address,
+      roles: ['Community member'],
+      points: 100,
+      date: ~~(new Date().getTime() / 1000),
+      profilePicture: await blobToBase64(imageFileObject)
+    }
+
+    let ipfsApiEndpointRequest
+    try {
+      ipfsApiEndpointRequest = await fetch(ipfsApiEndpoint, {
+        body: JSON.stringify(payLoad),
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        method: 'POST'
+      }).then((response) => {
+        // 413 crashes before we can check ipfsApiEndpointRequest.status,
+        // so it's dealt with here
+        if (response.status == 413) throw new Error(response.status.toString())
+        return response
+      })
+    } catch (error) {
+      processHttpError(413)
+      setIsLoading(false)
+      return
+    }
+    processHttpError(ipfsApiEndpointRequest.status)
+    const ipfsApiEndpointResponse = await ipfsApiEndpointRequest.json()
+    const tokenURI = await ipfsApiEndpointResponse.tokenUri
+
+    const pinataTokenUriRequest = await fetch(tokenURI)
+    processHttpError(pinataTokenUriRequest.status)
+    const responseJson = await pinataTokenUriRequest.json()
+    const tokenImageURI = await responseJson.image
+    if (tokenImageURI) {
+      setTokenImageURI(tokenImageURI)
+      return
+    }
+    setIsLoading(false)
+  }
+
+  const processHttpError = (status: number) => {
+    const httpErrorMsd: number = ~~(status / 100)
+    interface errorMessages {
+      [key: string]: string
+    }
+    const httpErrorMessages: errorMessages = {
+      error4: 'There was a problem with the request',
+      error400: 'There probably was an error with the data you submitted.',
+      error403: 'The server did not allow the request.',
+      error404: 'The server did not know what to do with the request.',
+      error413:
+        'The request payload was too large. Please choose an image below 30 megabytes.',
```

Comment:
```suggestion
        'The request payload was too large. Please choose an image below 40 kilobytes.',
```
I know 40 kilobytes is not much, but I think 150 kilobytes is an error.

Reply:
I was able to do 150 kilobytes...I guess that's the effect of the gateway fee.

Reply:
7 MB was fine for me and 8 MB wasn't... I agree on the 4 MB. I think recommending an image below 4 MB (but actually letting them post any size) should be reasonable. I don't think we'll be getting DOS attacks at this point :)